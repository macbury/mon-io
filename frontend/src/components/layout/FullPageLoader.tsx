import React from 'react'
import styled, { useTheme } from 'styled-components/native'
import { ActivityIndicator, View } from 'react-native'

const IndicatorContainer = styled(View)`
  flex: 1;
  justify-content: center;
  align-items: center;
  min-height: 320px;
  background: ${(props) => props.theme.windowBackground};
`

/**
 * Displays centered on screen activity indicator
 */
export default function FullPageLoader () {
  const theme = useTheme()
  return (
    <IndicatorContainer>
      <ActivityIndicator size={90} color={theme?.colors?.activityIndicator} />
    </IndicatorContainer>
  )
}