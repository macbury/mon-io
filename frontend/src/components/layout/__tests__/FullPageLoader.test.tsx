import React from 'react'
import { shallow } from 'enzyme'
import snapshot from 'enzyme-to-json'

import FullPageLoader from '../FullPageLoader'

it('renders correctly', () => {
  const component = shallow(<FullPageLoader />)
  expect(snapshot(component)).toMatchSnapshot()
})