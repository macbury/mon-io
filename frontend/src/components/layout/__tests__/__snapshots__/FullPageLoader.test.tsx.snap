// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Styled(View)>
  <ActivityIndicator
    size={90}
  />
</Styled(View)>
`;
